metrics/registry.go

```diff
@@ -0,0 +1,31 @@
+package metrics
+
+type registry struct {
+	provider Provider
+}
+
+//Provider interface that implements metric registry types
+type Provider interface {
+	Init()
+	Ingest(interface{})
```

Comment:
```suggestion
	ingest(interface{})
```
this too will be invoked via registry handler